### DISCUSSION TITLE
redis使用

### DISCUSSION THREAD

**Discussion by terry8210** [cloudwu/skynet#1663] [Q&A]
在使用skynet提供redis接口的时候，是否有必要把所有业务特性的redis操作到服务独立处理，通常情况下有一个dbproxy专业处理业务的数据库操作，是否需要有一个cacheproxy的存在呢？

**Comment by terry8210** [comment]
如果cacheproxy每次操作都需要发消息cacheproxy服务操作，会不会本末倒置呢？

**Comment by terry8210** [comment]
有没有同行遇到类似的问题？

**Comment by mingjava** [comment]
业务特性的redis操作是指的什么，我现在是一个独立的redis服务，这个服务就是与指定的redis配置建立连接，所有的redis相关的请求都发给这个redis服务。目前是用的多个redis实例，没有用redis的集群。

**Reply by terry8210** [reply]
我也是类似的做法，redis服务代理多个redis实例，所有命令操作发个服务区执行。

**Reply by mingjava** [reply]
那cacheproxy又是用来做什么的呢？

**Reply by terry8210** [reply]
cacheproxy就是你说的独立的redis服务

**Reply by mingjava** [reply]
嗯，那我们用法应该差不多，我是每个redis实例对应一个服务
